src/core/zowe/core_for_zowe_sdk/profile_manager.py

```diff
@@ -58,260 +71,169 @@ def config_appname(self) -> str:
             )
 
-        service_profile = {}
-
-        # load base profile
-        base_profile: dict = self.load_base_profile_properties(
-            profile_jsonc=profile_jsonc
+        service_profile: dict = {}
+        project_profile: dict = {}
+        global_profile: dict = {}
+        global_base_profile: dict = {}
+        project_profile_name: Union[str, None] = None
+        global_profile_name: Union[str, None] = None
+
+        # get Project Profile
+        project_profile, project_profile_name = self.get_profile(
+            self.project_config,
+            profile_name=profile_name,
+            profile_type=profile_type,
+            config_type="Project Config",
         )
-        service_profile.update(base_profile)
-
-        # load given profile
-        if profile_name is None:
-            profile_name = self.get_profilename_from_profiletype(
-                profile_jsonc=profile_jsonc, profile_type=profile_type
-            )
 
-        required_profile: dict = self.load_profile_properties(
-            profile_jsonc=profile_jsonc, profile_name=profile_name
+        # get Project User Profile
+        project_user_profile, project_user_profile_name = self.get_profile(
+            self.project_user_config,
+            profile_name=profile_name,
+            profile_type=profile_type,
+            config_type="Project User Config",
+        )
+        project_profile.update(project_user_profile)
+
+        # get Global Base Profile
+        global_base_profile, global_base_profile_name = self.get_profile(
+            self.global_config,
+            profile_name=None,
+            profile_type="base",
+            config_type="Global Config",
         )
-        service_profile.update(required_profile)
-
-        # apply profile args
-        if profile_args:
-            service_profile.update(profile_args)
-
-        return service_profile
-
-    def load_credentials(self) -> dict:
-        """
-        return credentials stored for the given config
-        """
-        credentials: dict = {}
-
-        try:
-            service_name = constants["ZoweServiceName"]
-
-            if sys.platform == "win32":
-                service_name += "/" + constants["ZoweAccountName"]
 
-            secret_value = keyring.get_password(
-                service_name, constants["ZoweAccountName"]
-            )
+        # get Global Base User Profile
+        global_base_user_profile, global_base_user_profile_name = self.get_profile(
+            self.global_user_config,
+            profile_name=None,
+            profile_type="base",
+            config_type="Global User Config",
+        )
+        global_base_profile.update(global_base_user_profile)
+
+        # get Global Profile
+        global_profile, global_profile_name = self.get_profile(
+            self.global_config,
+            profile_name=None,
+            profile_type="base",
+            config_type="Global Config",
+        )
```

Comment:
I believe the intention was to be able to load service profiles from the global config, but `base` was hardcoded 😢 
I tried loading a service profile from my global layers and none of them were loaded, only the base profile properties showed up 😋

Reply:
Thanks for finding the bug, I missed this while refactoring the code 😅 ... I will double check every other thing and address all of the issues